qiskit/optimization/algorithms/admm_optimizer.py

```diff
@@ -383,8 +385,17 @@ def solve(self, problem: QuadraticProgram) -> ADMMOptimizationResult:
         # third parameter is our internal state of computations.
         result = ADMMOptimizationResult(x=base_result.x, fval=base_result.fval,
                                         variables=base_result.variables,
-                                        state=self._state)
+                                        state=self._state,
+                                        status=base_result.status)
 
+        # check for feasibility
+        status = OptimizationResultStatus.SUCCESS if problem.is_feasible(result.x) \
+            else OptimizationResultStatus.INFEASIBLE
+
+        result = ADMMOptimizationResult(x=result.x, fval=result.fval,
```

Comment:
You don't need to create another `ADMMOptimizationResult`. You can check the feasibility as follows.
```python
base_result = int2bin.interpret(base_result)
status = OptimizationResultStatus.SUCCESS if problem.is_feasible(base_result.x) \
            else OptimizationResultStatus.INFEASIBLE
result = ADMMOptimizationResult(...)
```